docs/operations/azure_agents_en.md

```diff
@@ -0,0 +1,312 @@
+azureCredentials.setTenant(azureCredentialsTenantID);
+
+systemCredentialStore.addCredentials(domain, jenkinsAgentAdminCredentials)
+domainCredentials.add(azureCredentials);
+SystemCredentialsProvider.getInstance().setDomainCredentialsMap(domainCredentialsMap);
+
+
+
+// Configure Azure Cloud
+// See https://plugins.jenkins.io/azure-vm-agents/#plugin-content-configure-vm-template-using-groovy-script
+// See https://github.com/jenkinsci/azure-vm-agents-plugin/blob/master/src/main/java/com/microsoft/azure/vmagent/builders/AzureVMCloudBuilder.java
+def azureTemplate = new AzureVMTemplateBuilder()
+        .withName("jenkins-agent-azure")
+        .withLabels("docker node linux ubuntu")
+        .withLocation("West Europe")
+        .withVirtualMachineSize("Standard_DS3_v2")
+        .withExistingStorageAccount("jenkinsagentstor")
+        .addNewAdvancedImage()
+        .withCustomManagedImage("/subscriptions/93dbe1cb-8cce-42ab-a0bb-7b9759e50fc3/resourceGroups/ces-multinode_group/providers/Microsoft.Compute/images/cesJenkinsDockerAgent")
+        .withNetworkSecurityGroupName("Jenkins-Agents_NSG")
+        .endAdvancedImage()
+        .withAdminCredential(jenkinsAgentAdminCredentialsID)
+        .build()
+def azureCloud = new AzureVMCloudBuilder()
+        .withCloudName("Azure Cloud")
+        .withAzureCredentialsId(azureCredentialsID)
+        .withExistingResourceGroupName("ces_group")
+        .addToTemplates(azureTemplate)
+        .build()
+Jenkins.getInstance().clouds.add(azureCloud)
+
+```
+
+- Insert the Azure credentials from the steps before
+- Make sure the resource group name is correct
+- Run the script via the Jenkins script console or via the additional scripts directory as described above
+
+## Testing with a Jenkins pipeline
+
+You can try if everything worked out correctly by using this test-pipeline:
+- Add a Jenkins Pipeline project
+- Use the following code for the pipeline:
+
+```groovy
+#!groovy
+@Library(['github.com/cloudogu/ces-build-lib@1.62.0'])
+
+import com.cloudogu.ces.cesbuildlib.*
+
+node('docker') {
+    stage('Clone Repository') {
+        checkout scmGit(userRemoteConfigs: [
+            [url: 'https://github.com/cloudogu/sonar']
+        ])
+    }
+
+    def docker = new Docker(this)
+    stage('Docker Build') {
+        docker.withRegistry("https://my.registry.com", "registrycredentials") {
+            docker.build("testing/azuresonar:0.0.0")
+            sh 'docker image ls'
+        }
+    }
+
+    stage('Docker Tag') {
+        docker.image("testing/azuresonar:0.0.0").tag("0.0.1")
+        sh 'docker image ls'
+    }
+
+    stage('Docker Push') {
+        docker.withRegistry("https://my.registry.com", "registrycredentials") {
+            docker.image("testing/azuresonar:0.0.1").push()
+        }
+    }
+}
+
+```
+
+- Add Jenkins credentials with the ID `registrycredentials` to store your registry credentials
+- Adapt the registry URL to your container image registry
```

Comment:
````suggestion
    stage('Docker Run') {
        docker.image('hello-world').run('', '/hello')
    }
}

```

````
Maybe this easier example will suffice?

Reply:
I explicitly wanted to show that the `build`, `tag` and `push` functions are working, so I'll keep it as is.